File: checkout-core/src/main/java/com/adyen/checkout/core/PaymentResult.java
package com.adyen.checkout.core;

import android.os.Parcelable;
import android.support.annotation.NonNull;

import com.adyen.checkout.core.model.PaymentResultCode;

/**
 * The {@link PaymentResult} interface describes the result of a payment.
 * <p>
 * Copyright (c) 2018 Adyen B.V.
 * <p>
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 * <p>
 * Created by timon on 09/05/2018.
 */
public interface PaymentResult extends Parcelable {
    /**
     * @return The {@link PaymentResultCode}.
     */
    @NonNull
    PaymentResultCode getResultCode();

    /**
     * @return The payload that can be used to retrieve further information about the result of the payment.
     * @see <a href="https://docs.adyen.com/developers/checkout/android-sdk/quick-start-android/verify-payment-result-android">
     *     Verifying a payment result</a>
     */
    @NonNull
    String getPayload();
}

File: core/src/main/java/com/adyen/checkout/core/common/PaymentResult.kt
/*
 * Copyright (c) 2025 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by ozgur on 15/8/2025.
 */

package com.adyen.checkout.core.common

import android.os.Parcelable
import com.adyen.checkout.core.components.data.OrderResponse
import kotlinx.parcelize.Parcelize

// TODO - Kdocs
@Parcelize
data class PaymentResult(
    val resultCode: String,
    val sessionId: String?,
    val sessionResult: String?,
    // TODO - Check whether sessionData is required or not?
    val sessionData: String?,
    // TODO - Check whether order is required or not
    val order: OrderResponse?,
) : Parcelable
